### ISSUE TITLE
RFC: Evaluate a mask-aware teacher for difficult waste annotation cases

### ISSUE THREAD

**Issue by WhiteStoneTak** [ryuto1127/wayste#9]
## Context

Wayste already has a thoughtful teacher-student workflow in
`scripts/train/autolabel_frames.py`:

```text
Grounding DINO → human QA → YOLO training
```

Before proposing a specific implementation, I would like to ask whether the
current box-based auto-labeling workflow has difficulty with cases such as:

1. A hand overlapping the waste item
2. Transparent containers blending into the background
3. Connected components, such as a cup and lid
4. Irregularly shaped bags and wrappers
5. Multiple items appearing in one frame

These cases may produce loose bounding boxes, merge separate components, or
increase the amount of manual correction required. However, I have not yet
benchmarked them against Wayste's current training data, so I would first like
to confirm whether they are actual pain points for the project.

## Hypothesis

A mask-aware teacher such as
[[Meta SAM 3](https://github.com/facebookresearch/sam3)](https://github.com/facebookresearch/sam3) may improve pseudo-label
localization and QA for these cases.

SAM 3 can return instance masks from text or visual exemplar prompts. The masks
could be used to:

- Derive tighter YOLO bounding boxes
- Separate the item from an overlapping hand or background
- Identify multiple instances
- Visualize object boundaries in QA sheets
- Reject masks that are too small or cover most of the frame

SAM 3 would only be used during offline dataset preparation. The browser and
kiosk runtime would continue using the existing YOLO model.

## Proposed first step

Rather than adding SAM 3 immediately, I propose a small comparison experiment:

```text
Same manually verified images
  ├── Grounding DINO teacher
  └── SAM 3 teacher
          ↓
Compare generated labels and human correction effort
```

Possible evaluation criteria:

- Auto-accepted label precision
- Human correction rate
- No-detection and uncertain rates
- Bounding-box quality
- Multi-item recall
- Downstream YOLO performance on a separate human-labeled test set

Model-specific confidence thresholds would be calibrated separately.

## Implementation direction if the experiment is successful

If SAM 3 shows a measurable benefit, the existing auto-labeling script could
support selectable teacher backends:

```bash
python scripts/train/autolabel_frames.py frames dataset \
  --teacher grounding-dino

python scripts/train/autolabel_frames.py frames dataset \
  --teacher sam3
```

Grounding DINO would remain the default. SAM 3 would be an optional external
dependency, with no source code or model weights committed to this repository.

DART or other acceleration layers would remain out of scope until SAM 3's
annotation benefit is demonstrated and inference speed becomes a real
bottleneck.

## Questions

1. Are the five cases above observed problems in the current auto-labeling or
   training workflow?
2. Is there a representative, manually verified image set that could be used
   for a comparison?
3. Would a small benchmark-first contribution be useful before proposing an
   implementation PR?
